### PR TITLE
chore: booker atom username / teams props enabled queries

### PR DESCRIPTION
**Pull request by ThyMinimalDev**
## What does this PR do?

Booker Atom prop username was mandatory even for team event types, created discriminative union to show which props are required depending on team / individual event-type. slots controller does not require usernames anymore.

improve react query hooks to avoid non-necessary fetching of data

https://github.com/user-attachments/assets/c10a2e65-f3ae-4abd-9c74-6060beabd155

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

use cal.com booker atom with team event-types

### CODE REVIEW

File: packages/platform/atoms/booker/BookerPlatformWrapper.tsx [hunk 1]
```diff
@@ -73,11 +75,24 @@ export type BookerPlatformWrapperAtomProps = Omit<BookerProps, "username" | "ent
...
 
-export const BookerPlatformWrapper = (props: BookerPlatformWrapperAtomProps) => {
+type BookerPlatformWrapperAtomPropsForIndividual = BookerPlatformWrapperAtomProps & {
+  username: string | string[];
+  isTeamEvent?: false;
+};
+
+type BookerPlatformWrapperAtomPropsForTeam = BookerPlatformWrapperAtomProps & {
+  username?: string | string[];
+  isTeamEvent: true;
```
Comment (supalarry) [inline packages/platform/atoms/booker/BookerPlatformWrapper.tsx:87]: Idea for future - I think we can simplify by inferring that if `teamId` is passed to set `isTeamEvent` to true.

File: packages/platform/atoms/booker/BookerPlatformWrapper.tsx [hunk 2]
```diff
@@ -107,23 +122,28 @@ export const BookerPlatformWrapper = (props: BookerPlatformWrapperAtomProps) =>
...
     isError: isTeamError,
     isPending: isTeamPending,
-    data: teamData,
-  } = useTeamEventType(props.teamId, props.eventSlug, props.isTeamEvent);
+    data: teamEventTypeData,
+  } = useTeamEventType(teamId, props.eventSlug, props.isTeamEvent);
 
   const event = useMemo(() => {
-    if (props.isTeamEvent) {
+    if (props.isTeamEvent && !isTeamPending && teamId && teamEventTypeData && teamEventTypeData.length > 0) {
```
Comment (supalarry) [inline packages/platform/atoms/booker/BookerPlatformWrapper.tsx:139]: Why adding the checks `&& !isTeamPending && teamId && teamEventTypeData && teamEventTypeData.length > 0` ?

On line 145 we already check for data transformation and if the team is pending (loading) then we don't enter if statement and return non team response.

File: packages/platform/atoms/hooks/event-types/public/useEventType.ts [hunk 3]
```diff
@@ -13,21 +11,18 @@ import http from "../../../lib/http";
 export const QUERY_KEY = "use-event-type";
 export type UsePublicEventReturnType = ReturnType<typeof useEventType>;
 
-export const useEventType = (username: string, eventSlug: string, isTeamEvent: boolean | undefined) => {
-  const [stateUsername, stateEventSlug] = useBookerStore(
-    (state) => [state.username, state.eventSlug],
-    shallow
-  );
+export const useEventType = (username: string, eventSlug: string, isTeamEvent: boolean | undefined, teamId?: number) => {
```
Comment (supalarry) [inline packages/platform/atoms/hooks/event-types/public/useEventType.ts:14]: teamId is not used anywhere and can be removed from params

File: packages/platform/atoms/booker/BookerPlatformWrapper.tsx (hunk 1 above)
Comment (ThyMinimalDev) [reply packages/platform/atoms/booker/BookerPlatformWrapper.tsx:87]: good idea

File: packages/platform/atoms/booker/BookerPlatformWrapper.tsx (hunk 2 above)
Comment (ThyMinimalDev) [reply packages/platform/atoms/booker/BookerPlatformWrapper.tsx:139]: if the teamId changes, we don't want to render previous data while it's fetching new one

**Review by supalarry — CHANGES_REQUESTED** [review]
Tried examples base app and when i click on Book me -> then on collective event prisma errors in v2 api

Screenshot 2024-09-06 at 10 59 21

it is team event-type so it should have landed in organization event-types controller and idk why id is a string maybe was already before

**Comment by supalarry** [comment]
> Tried examples base app and when i click on Book me -> then on collective event prisma errors in v2 api
> 
> Screenshot 2024-09-06 at 10 59 21
> it is team event-type so it should have landed in organization event-types controller and idk why id is a string maybe was already before

Yeah its this PR - just built atoms on main and collective event type booker loads fine:
Screenshot 2024-09-06 at 11 06 11

with this PR it said no availability and had v2 error above

**Review by supalarry — DISMISSED** [review]
rechecked again locally and works

**Review by zomars — APPROVED** [review]

**Pull request merged**